linters/amp_fix_md_links.py

```diff
@@ -63,6 +63,38 @@ def _make_path_module_agnostic(path: str) -> str:
     return upd_path
 
 
+def _check_md_section_exists(
+    link_in_cur_module: str,
+    section: str,
+    link_path: str,
+    file_name: str,
+    line_num: int,
+) -> str | None:
+    """
+    Check if a section or heading exists in the markdown file.
+
+    Searches the specified file for a Markdown-style section that
+    matches the provided section name. Returns a warning message if the
+    section is not found.
+
+    :param link_in_cur_module: the path to the Markdown file to check
+    :param section: the heading text to look for in the file
+    :param file_name: the original file with the link
+    :param line_num: the number of the line in the file
+    :return: a warning message if the referenced markdown section is
+        missing, or None if found
+    """
+    with open(link_in_cur_module, "r", encoding="utf-8") as file:
+        content = file.read()
+    section_pattern = re.compile(r"\[.*?\]\((.*?)\)", re.MULTILINE)
+    matches = section_pattern.findall(content)
+    if f"#{section}" in matches:
+        # Return None if the specified section is found.
+        return None
+    warning = f"{file_name}:{line_num}: The section '{section}' does not exist in the '{link_path}'"
```

Comment:
You are passing extra stuff to build the warning. But this is not competence of this function.
This function should only return yes/no, then the caller decides what to do